File: Quiz1/Kingsman/Part.cs
namespace Quiz1.Kingsman;

public class Part
{
    private World world;


    public bool Run()
    {
        var teenager = new Teenager();

        teenager.Teach();
        if (teenager.Test()) return true;

        world.SaveBy(teenager);

        return teenager.IsGentleman();
    }
}

public class Teenager
{
    public void Teach()
    {
    }


    public bool Test()
    {
        return true;
    }

    public bool IsGentleman()
    {
        return true;
    }
}

public class World
{
    public void SaveBy(Teenager teenager)
    {
    }
}

File: Quiz1/Kingsman/Part.fs
type Part() =
    let world = World()

    member this.Run() =
        let teenager = Teenager()

        teenager.Teach()
        if teenager.Test() then return true

        world.SaveBy(teenager)

        teenager.IsGentleman()

type Teenager() =
    member this.Teach() = ()

    member this.Test() = true

    member this.IsGentleman() = true

type World() =
    member this.SaveBy(teenager: Teenager) = ()